tests/real/enterprise.rs

```diff
@@ -370,3 +391,693 @@ fn ent_unique_job_bypass_unique_lock() {
+    // Let's see if we will be able to - again - open the committed batch "from outside" and
+    // add a nested batch to it.
+    let mut b = p.open_batch(bid.clone()).unwrap().expect("is some");
+    let mut nested_callbacks = some_jobs(
+        "order_clean_up__NESTED",
+        "test_batch_can_be_reopned_add_extra_jobs_added__CALLBACKs__NESTED",
+        2,
+    );
+    let nested_batch_declaration = Batch::builder()
+        .description("Orders processing workload. Nested stage")
+        .with_callbacks(
+            nested_callbacks.next().unwrap(),
+            nested_callbacks.next().unwrap(),
+        );
+    let nested_batch = b.start_batch(nested_batch_declaration).unwrap();
+    let nested_bid = nested_batch.id().to_string();
+    // committing the nested batch without any jobs
+    // since those are just not relevant for this test:
+    nested_batch.commit().unwrap();
+
+    let s = t.get_batch_status(nested_bid.clone()).unwrap().unwrap();
+    assert_eq!(s.total, 0);
+    assert_eq!(s.parent_bid, Some(bid)); // this is really our child batch
+    assert_eq!(s.complete_callback_state.to_string(), "Enqueued");
+
+    // Subtest 3 result:
+    // We managed to open an already committed batch "from outside" and the server accepted
+    // a nested batch INSTEAD OF ERRORING BACK.
+    // ############################ END OF SUBTEST 3 #######################################
+
+    // The following subtest assertions should be adjusted once fixes are introduced to
+    // the Faktory as per https://github.com/contribsys/faktory/issues/465
+    // The idea is we should not be able to push to a batch for which the server have already
+    // enqeued a callback.
+    //
+    // ############################## SUBTEST 4 ############################################
+    // From the docs:
+    // """Once a callback has enqueued for a batch, you may not add anything to the batch."""
+    // ref: https://github.com/contribsys/faktory/wiki/Ent-Batches#guarantees (Jan 10, 2024)
+
+    // Let's try to re-open the nested batch that we have already committed and add some jobs to it.
+    let mut b = p
+        .open_batch(nested_bid.clone())
+        .expect("no error")
+        .expect("is some");
+    let mut more_jobs = some_jobs(
+        "order_clean_up__NESTED",
+        "test_batch_can_be_reopned_add_extra_jobs_added__NESTED",
+        2,
+    );
+    b.add(more_jobs.next().unwrap()).unwrap();
+    b.add(more_jobs.next().unwrap()).unwrap();
+    b.commit().unwrap();
+
+    let s = t.get_batch_status(nested_bid.clone()).unwrap().unwrap();
+    match s.complete_callback_state {
+        CallbackState::Enqueued => {}
+        _ => panic!("Expected the callback to have been enqueued"),
+    }
+    assert_eq!(s.pending, 2); // ... though there are pending jobs
+    assert_eq!(s.total, 2);
+
+    // Subtest 4 result:
+    // We were able to add more jobs to the batch for which the Faktory server had already
+    // queued the callback.
```

Comment:
The other option would be to not make any assertions here, but since this is a documented guarantee, I decided to leave this subtest as-is and as soon as the fixes are introduced - to update the test.